core/domain/state_domain.py

```diff
@@ -1200,16 +1222,25 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
                 raise utils.ValidationError(
                     'Found a duplicate content id %s' % solution_content_id)
             content_id_list.append(solution_content_id)
-        available_content_ids = self.content_ids_to_audio_translations.keys()
-        if not set(content_id_list) <= set(available_content_ids):
+        available_content_ids_in_audio_translations = (
+            self.content_ids_to_audio_translations.keys())
+
+        self.content_translations.validate(
+            expected_content_id_list=content_id_list)
+
+        if not set(content_id_list) == set(
```

Comment:
Are there checks to validate that all elements in content_id_list are distinct? If not we should add such checks (and tests for them).